bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/impl/ProductGraphqlServicImpl.java

```diff
@@ -0,0 +1,151 @@
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ *
+ *GraphQlService extracted from ProductModel
+ * It could be one service to cater to all component , carrying all the service methods
+ * ToDO: Refactoring
+ *
+ */
+
+@Component(service = ProductGraphqlService.class)
+public class ProductGraphqlServicImpl implements ProductGraphqlService {
+
+    private static final String PRODUCT_IMAGE_FOLDER = "catalog/product";
+    private String mediaBaseUrl;
+
+
+    @Override
+    public List<Asset> filterAndSortAssets(List<MediaGalleryEntry> assets) {
+        return assets.parallelStream()
+                .filter(e -> !e.getDisabled() && e.getMediaType().equals("image"))
+                .map(this::mapAsset)
+                .sorted(Comparator.comparing(Asset::getPosition))
+                .collect(Collectors.toList());
+    }
+    @Override
+    public ProductInterface getProduct(GraphqlClient client, String sku  ) {
+        FilterTypeInput input = new FilterTypeInput().setEq(sku);
+        ProductFilterInput filter = new ProductFilterInput().setSku(input);
+        QueryQuery.ProductsArgumentsDefinition searchArgs = s -> s.filter(filter);
+        ProductsQueryDefinition queryArgs = q -> q.items(this.generateProductQuery());
+        String queryString = Operations.query(query -> query
+                .products(searchArgs, queryArgs)
+                .storeConfig(this.generateStoreConfigQuery())).toString();
+        GraphqlResponse<Query, Error> response = client.execute(new GraphqlRequest(queryString), Query.class, Error.class, QueryDeserializer.getGson());
+        Query rootQuery = response.getData();
+        List<ProductInterface> products = rootQuery.getProducts().getItems();
+        mediaBaseUrl = rootQuery.getStoreConfig().getSecureBaseMediaUrl();
+        if (products.size() > 0) {
+            return products.get(0);
+        }
+        return null;
+    }
+
+    public ProductPricesQueryDefinition generatePriceQuery() {
+        return q -> q
+                .regularPrice(rp -> rp
+                        .amount(a -> a
+                                .currency()
+                                .value()));
+    }
+
+    public SimpleProductQueryDefinition generateSimpleProductQuery() {
+        return q -> q
+                .id()
+                .sku()
+                .name()
+                .description(d -> d.html())
+                .image(i -> i.label().url())
+                .thumbnail(t -> t.label().url())
+                .urlKey()
+                .stockStatus()
+                .color()
+                .price(this.generatePriceQuery())
+                .mediaGalleryEntries(g -> g
+                        .disabled()
+                        .file()
+                        .label()
+                        .position()
+                        .mediaType());
+    }
+
+    public ProductInterfaceQueryDefinition generateProductQuery() {
```

Comment:
I think the query for this component (independently from the ProductGraphqlService topic) can be simplified very much, many of these attributes are not needed for the teaser. That is actually the power of GraphQL that one can only request the data relly needed.